src/storage.rs

```diff
@@ -1,8 +1,42 @@
-pub struct FileStorage {}
+use miniserde::{json, Deserialize, Serialize};
+use std::fs;
+use std::path::Path;
+
+const FILENAME: &str = "filestorage.json";
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FileStorage {
+    username: String,
+    password: String,
+    token: String,
+    course: String,
+    task: String,
+    language: String,
+    option: String,
+    file: String,
+}
 
 impl FileStorage {
     pub fn new() -> Self {
-        Self {}
+        let empty = FileStorage {
+            username: String::from(""),
+            password: String::from(""),
+            token: String::from(""),
+            course: String::from(""),
+            task: String::from(""),
+            language: String::from(""),
+            option: String::from(""),
+            file: String::from(""),
+        };
```

Comment:
For helping in constructing this empty value, you can use the `Default` trait. It will make all string fields empty by default:

```rust
#[derive(Default, Serialize, Deserialize, Debug)]
struct FileStorage {
    username: String,
    ...
}

let empty = FileStorage::default();
```